src/sentry/rules/processing/delayed_processing.py

```diff
@@ -92,42 +100,75 @@ def get_rules_to_slow_conditions(
     return rules_to_slow_conditions
 
 
-def get_condition_groups(
+def _generate_unique_queries(
+    condition_data: EventFrequencyConditionData, environment_id: int
+) -> list[UniqueConditionQuery]:
+    """
+    Returns a list of all unique condition queries that must be made for the
+    given condition instance.
+    Count comparison conditions will only have one unique query, while percent
+    comparison conditions will have two unique queries.
+    """
+    unique_queries = [
+        UniqueConditionQuery(
+            cls_id=condition_data["id"],
+            interval=condition_data["interval"],
+            environment_id=environment_id,
+        )
+    ]
+    if condition_data.get("comparisonType") == ComparisonType.PERCENT:
+        # We will later compare the first query results against the second query to calculate
+        # a percentage for percentage comparison conditions.
+        comparison_interval = condition_data.get("comparisonInterval", DEFAULT_COMPARISON_INTERVAL)
+        second_comparison_query = unique_queries[0]._replace(comparisonInterval=comparison_interval)
+        unique_queries.append(second_comparison_query)
```

Comment:
Hmm - something about this feels a little off. it seems like we're creating a unique query, modifying it, and then appending it to the list.

I'm a little nervous about the mutation to the [unique_queries[0]._replace](https://github.com/getsentry/sentry/blob/bc54bac083ab7c585b778e8e1686cc274d8d2de8/src/sentry/rules/processing/delayed_processing.py#L123) since it reads like a mutation of the first element. While `._replace` creates a copy rather than mutating, maybe we help our future selves by making a copy of the query and then updating it?

Reply:
agreed with this.
Can we simple construct another class instance and splat in the values while overwriting the comparion interval instead?

Reply:
I was following the docs [here](https://docs.python.org/dev/library/collections.html#collections.somenamedtuple._replace) but for readability I can explicitly make a new instance

Reply:
ok I can't create a copy and then update it's value cuz it's a tuple :|
I can use unpacking but it's a bit ugly as it doesn't handle overrides.

```Python
first_query_copy = unique_queries[0]._asdict()
first_query_copy["comparison_interval"] = comparison_interval
second_comparison_query = UniqueConditionQuery(**first_query_copy)
```